aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -215,7 +258,8 @@ Tensor zeros_like(const Tensor& self) {
 Tensor zeros_like(const Tensor& self, const Type& dtype) {
   if (dtype.is_sparse() && self.type().is_sparse()) {
     auto res = dtype.tensor();
-    res.resize_as_(self);
+    // resize_as_ requires the same exact type.
+    res.sparse_raw_resize_(self.sizes(), self._dimI(), self._dimV());
     res.zero_();
```

Comment:
Is the `res.zero_()` call necessary for sparse tensors?

Reply:
No, I can get rid of it.